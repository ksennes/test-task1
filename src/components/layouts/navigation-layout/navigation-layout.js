import React, { useState } from "react";
import { Link } from 'react-router-dom';

import "./navigation-layout.scss";
import logo from "../../../img/logo.png";
import greenLogo from '../../../img/logo1.png';
import burger from '../../../img/menu.png';
import cross from '../../../img/cross.png';
import filter from '../../../img/filter.png';
import activeFilter from '../../../img/filterActive.png';
import { DesktopNavigation } from "./desktop-navigation/desktop-navigation";
import { MobileNavigation } from "./mobile-navigation/mobile-navigation";

export const NavigationLayout = ({ isLogged, activeLink, isFilterOpen, handleFilter }) => {
  const navigationList = [
    { title: "JOGS", link: "/jog" },
    { title: "INFO", link: "/info" },
    { title: "CONTACT US", link: "#"},
  ];

  const [isOpen, setIsOpen] = useState(false);

  const handleOpenMenu = () => {
      setIsOpen(true);
  }

  const handleCloseMenu = () => {
      setIsOpen(false);
  }

  return (
    <div className="navigation-layout" style={isOpen? {backgroundColor:'#fff'} : null}>
      <div className="navigation-layout__logo">
        <img src={isOpen? greenLogo : logo} />
      </div>
      <div className="navigation-layout__options">
        {isLogged ? (
            <>
            <DesktopNavigation navigationList={navigationList} activeLink={activeLink}/>
            <div className='filter__icon'>
              <img src={isFilterOpen ? activeFilter : filter} onClick={handleFilter}/>
            </div>
            <div className='burger' onClick={() => handleOpenMenu()} style={isOpen ? {display: 'none'} : null}>
                <img src={burger}/>
            </div>
            <div className='cross' onClick={() => handleCloseMenu()} style={isOpen ? null: {display: 'none'}}>
                <img src={cross}/>
            </div>
            <MobileNavigation navigationList={navigationList} activeLink={activeLink} isOpen={isOpen}/>
            </>
        ) : (
          ""
        )}
      </div>
    </div>
  );
};
