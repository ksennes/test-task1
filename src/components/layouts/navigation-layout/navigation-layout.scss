.navigation-layout {
  width: 100%;
  height: 116px;
  background-color: #7ed321;
  display: flex;
  align-items: center;
  justify-content: space-between;

  &__logo {
    margin-left: 37px;

    .mobile-logo {
      display: none;
    }
  }

  &__options {
    display: flex; 
    align-items: center;

    .filter__icon {
      margin-right: 36px;
      margin-left: 56px;

      
    }

    .burger,
    .cross {
      display: none;
      margin-right: 25px;
    }
  }
}

@media (max-width: 960px) {
  .navigation-layout {
    height: 77px;

    &__logo {
      height: 37px;
      margin-left: 25px;
    }

    &__options {
      .burger,
      .cross {
        display: block;
      }
    }
  }
}
