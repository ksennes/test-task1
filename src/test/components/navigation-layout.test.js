import React from 'react';
import { NavigationLayout } from '../../components/layouts/navigation-layout/navigation-layout';
import renderer from 'react-test-renderer';

describe('NavigationLayout component', () => {
  it('sould match the last snapshot without props', () => {
    const wrapper = renderer.create(<NavigationLayout />)
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('sould match the last snapshot with isLogged prop', () => {
    const props = {
        isLogged: false, 
    }
    const wrapper = renderer.create(<NavigationLayout props={props}/>)
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('sould match the last snapshot with props', () => {
    const handler = jest.fn();
    const props = {
        isLogged: false, 
        activeLink: '/jogs', 
        isFilterOpen: false, 
        handleFilter: handler
    }
    const wrapper = renderer.create(<NavigationLayout props={props}/>)
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});