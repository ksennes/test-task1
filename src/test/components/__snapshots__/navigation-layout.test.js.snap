// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLayout component sould match the last snapshot with isLogged prop 1`] = `
<div
  className="navigation-layout"
  style={null}
>
  <div
    className="navigation-layout__logo"
  >
    <img
      src="test-file-stub"
    />
  </div>
  <div
    className="navigation-layout__options"
  >
    
  </div>
</div>
`;

exports[`NavigationLayout component sould match the last snapshot without props 1`] = `
<div
  className="navigation-layout"
  style={null}
>
  <div
    className="navigation-layout__logo"
  >
    <img
      src="test-file-stub"
    />
  </div>
  <div
    className="navigation-layout__options"
  >
    
  </div>
</div>
`;
